common/src/BasePlugin.php

```diff
@@ -103,19 +103,60 @@ public function isWoocommerceActivated(): bool {
         return class_exists('woocommerce');
     }
 
+    public function getActiveGtinPlugin() {
+        return GtinHandler::getActivePlugin();
+    }
+
     public function getProductMetaKeys(): array {
         global $wpdb;
-        return $wpdb->get_col("
+
+        $meta_keys = $wpdb->get_col("
             SELECT DISTINCT(pm.meta_key)
             FROM {$wpdb->posts} p
             LEFT JOIN {$wpdb->postmeta} pm ON p.ID = pm.post_id
             WHERE
                 p.post_type = 'product'
                 AND pm.meta_key <> ''
         ");
+        return array_merge($meta_keys, $this->getCustomAttributes());
+    }
+
+    private function getCustomAttributes(): array {
+        global $wpdb;
+        $custom_attributes = [];
+        $sql = "
+            SELECT meta.meta_id, meta.meta_key as name, meta.meta_value as type 
+            FROM {$wpdb->postmeta} meta
+            JOIN {$wpdb->posts} posts
+            ON meta.post_id = posts.id 
+            WHERE posts.post_type LIKE '%product%' 
+            AND meta.meta_key='_product_attributes';";
+
+        $data = $wpdb->get_results($sql);
+        if (!empty($data)) {
```

Comment:
>(array|object|null) Database query results.

I am always confused with these returns from Wordpress.

Reply:
> > (array|object|null) Database query results.
> 
> I am always confused with these returns from Wordpress.

It probably returns non-array values in case of errors.  Empty result should be an empty array, I'd be surprised otherwise.